enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+the baremetal IPI installer or ZTP APIs use to describe the cluster
+(name, API IPs, etc.) and host-specific network settings (static IPs,
+VLANs, etc.). Using the existing file and custom resource types allows
+us to use the same version management techniques that we do for the
+APIs within a cluster.
+
+The `create image` command will produce a single image as output, to
+save storage space and reduce management complexity. The assisted
+installer service can already generate images with boot scripts that
+recognize which host is booting and apply host-specific network
+settings and roles (such as running different services) accordingly,
+and we will reuse that technique (and code).
+
+One host, designated as "node0", will be configured to run the
+assisted service and a client that uses the assisted service API to
+automate the deployment.
+
+All other hosts will run the assisted agent, configured to talk to the
+service on node0.
+
+When all of the images are booted, the client on node0 will use the
+assisted service to trigger the deployment of the cluster (including
+applicable pre-flight validations) without further interaction from
+the user.
+
+The `create image` command will produce as output credentials to
+access the cluster that is about to be created. This allows a client,
+such as a tool monitoring deployment progress, to use the cluster's
+API as soon as the API service is available.
+
+During deployment, the third-party orchestration tool will need a way
+to monitor progress and discover errors. The assisted service
+read-only API endpoints running on node0 will report that data. In the
+event of failure, a read-only endpoint in that service will be
+available to download any logs that have been collected.
+
+We will be taking advantage of the assisted service's ability to run
+the bootstrap services "in place" to avoid having a separate bootstrap
+VM. Bootstrap will run on node0, where the assisted service runs, to
+ensure that the service and its API are available for as long as
+possible. When bootstrapping completes and node0 reboots, progress
+monitoring will need to shift to the OpenShift APIs (`ClusterVersion`
+and `ClusterOperator`). We will extend the `openshift-install wait-for
+install-complete` command to recognize when the cluster being deployed
+uses the agent installer and to use both APIs for monitoring and
+[progress
+reporting](https://github.com/openshift/assisted-service/blob/master/docs/enhancements/installation-progress-bar.md)
+in that case, including reporting when steps like validation fail and
+the cluster deployment is never going to succeed. Users will still be
+able to write their own monitoring logic using the assisted service
+and OpenShift APIs directly, but they will have the option of using
+the command line tool.
+
+For the fully automated workflow, the steps a user might normally
+perform such as approving hosts or triggering the installation will be
```

Comment:
@staebler there is not create cluster stage in AI. only create manifests. And then those ignitions are propagated to the nodes

Reply:
@staebler in the agent-based install pattern, the agents each phone home to the central service, send hardware details, do some validation, and then wait. They wait for the central service to tell them to start forming a cluster, and that service also tells each whether it will be the bootstrap host or not along with other critical info. The client described here is just performing that action of initiating the installation.

Reply:
Perhaps we need to define more specific terms so that the term "installation" is not so overloaded.

Reply:
>There's definitely some, uh, finessing to do to be able to use creds pre-generated and stored in the ISO before assisted runs openshift-install create manifests on the bootstrap smile

Yes. This is a very important gap that we must cover. To have the admin cred generated in ISO creation stage be valid after when assisted-service runs openshift-install again. This is currently far from existing.